workshops/meme-generator/README.md

```diff
@@ -0,0 +1,385 @@
+
+    // Top text font size
+    let fontSize = canvas.width * topTextSize; //Font Size will change based on our input sliders
+    ctx.font = `${fontSize}px Impact`; // We'll be using Impact font, which is used by most memes
+    ctx.lineWidth = fontSize / 20; // lineWidth will be the outline of our text, and we're setting it to be 20th of our fontSize here.
+
+    // Draw top text
+    ctx.textBaseline = 'top'; // textBaseline property specifies the current text baseline used when drawing text.
+    topText.split('\n').forEach((t, i) => {
+      ctx.fillText(t, canvas.width / 2, i * fontSize, canvas.width); // fillText takes 3 arguments: first is our text, second and third arguments are our X and Y coordinates of the point at which to begin drawing the tex.
+      ctx.strokeText(t, canvas.width / 2, i * fontSize, canvas.width); // Arguments are same as fillText but strokeText draws outlines on our text.
+    });
+
+    // Bottom text font size
+    fontSize = canvas.width * bottomTextSize;
+    ctx.font = `${fontSize}px Impact`;
+    ctx.lineWidth = fontSize / 20;
+
+    // Draw bottom text
+    ctx.textBaseline = 'bottom';
+    bottomText.split('\n').reverse().forEach((t, i) => { // .reverse() because it's drawing the bottom text from the bottom up
+      ctx.fillText(t, canvas.width / 2, canvas.height - i * fontSize, canvas.width);
+      ctx.strokeText(t, canvas.width / 2, canvas.height - i * fontSize, canvas.width);
+    });
+  }
+    ```
+
+  This code block is really simple:
+
+  1. Inside the `generateMeme()` function we initialise a [`Canvas`](https://developer.mozilla.org/en-US/docs/Web/API/Canvas_API/Tutorial/Basic_usage) which .
+  2. To get the canvas' [2D rendering context](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D), we call `getContext()` by selecting out `<canvas>` element from HTML by its ID, and supplying `'2d'` as the argument. Our 2D rendering context will provide us with a lot of methods to stylize and draw on our canvas!
+  3. Next, we set the canvas' dimensions same as our meme image's dimensions.
+  4. Then we clear, and draw the image on the canvas.
+  5. Next, to draw text on our meme image, we set the style of our text and drawing the text using `ctx.fillText` and `ctx.strokeText` for outlines, where `ctx` is our [2D rendering context](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D).
+
+* Now, after our `generateMeme` function, add a [Window: DOMContentLoaded](https://developer.mozilla.org/en-US/docs/Web/API/Window/DOMContentLoaded_event) event listener which will listen to `DOMContentLoaded` event, which will be executed after the content of the web page is loaded and which will also call our `generateMeme` function.
+
+    ```javascript
+    window.addEventListener('DOMContentLoaded', () => {
+        //Code here
+    });
+    ```
+
+  Inside the event, add the following code:
+
+  ```javascript
+  window.addEventListener('DOMContentLoaded', () => {
```

Comment:
I know you explain what this code block does later, but I don't think people will fully understand because it's such a big code block and people will just copy/paste without understanding. I think you should break it down